src/ucp/core/ucp_am.c

```diff
@@ -70,6 +70,41 @@ void ucp_am_ep_cleanup(ucp_ep_h ep)
     }
 }
 
+size_t ucp_am_max_header_size(ucp_worker_h worker)
+{
+    uct_iface_attr_t *if_attr;
+    ucp_rsc_index_t iface_id;
+    size_t max_am_header, max_uct_header;
+
+    if (!(worker->context->config.features & UCP_FEATURE_AM)) {
+        return 0ul;
+    }
+
+    max_am_header = SIZE_MAX;
+
+    for (iface_id = 0; iface_id < worker->num_ifaces; ++iface_id) {
+        if_attr = &worker->ifaces[iface_id]->attr;
+        if (if_attr->cap.flags & UCT_IFACE_FLAG_AM_BCOPY) {
+            max_am_header = ucs_min(max_am_header, if_attr->cap.am.max_bcopy);
+        }
+
+        if (if_attr->cap.flags & UCT_IFACE_FLAG_AM_ZCOPY) {
+            max_am_header  = ucs_min(max_am_header, if_attr->cap.am.max_zcopy);
```

Comment:
why is this done, the next comment says that in case of zcopy the user header is placed into am_hdr?
also, if user header is larger than am_hdr, can we send it with am_bcopy?